### DISCUSSION TITLE
How to convert code for esp-idf to arduino

### DISCUSSION THREAD

**Discussion by dagi0627** [espressif/arduino-esp32#9624] [General]
i want to convert code native_ota_example.c in esp-idf to arduino code. How to do it ?

**Comment by yunyizhi** [comment]
*  only one  `void app_main(void) `
 void app_main(void)  -> void app_main() 
* call app_main()
void setup() {
app_main();
}
void loop() {
  delay(10 * 1000);
}
* `ESP-IDF LOG`  set level by  Menu Core Debug Level
 platformio.ini as 
 ```ini
    build_flags =
    -D CONFIG_ARDUHAL_LOG_COLORS=1
    -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_INFO
  ```
there is the esp-idf cdc_acm_vcp example use by arduino : https://gitee.com/yunyizhi/espressif-demo/tree/master/cdc_acm_vcp
